src/charm.py

```diff
@@ -683,6 +773,19 @@ def _routing_mode(self) -> _RoutingMode:
         """
         return _RoutingMode(self.config["routing_mode"])
 
+    @property
+    def cert_subject(self) -> str:
+        """Provide certificate subject."""
+        # TODO: Use a good default for when the external_hostname is None
+        return self.model.config.get("external_hostname", "whatever")
```

Comment:
Iiuc, the only concern with `self.external_url` is that it may return an IP address.

If that is the case, would this work?

```suggestion
        return self.model.config.get("external_hostname", socket.getfqdn())
```

cc: @rbarry82